### PR TITLE
adding Delsarte-Goethals, Goethals, Nordstrom-Robinson codes

### PR DESCRIPTION
**Pull request by MadhuraPankaja**

### CODE REVIEW

File: codes/classical/bits/delsarte_goethals.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+
+code_id: delsarte_goethals
+physical: bits
+logical: bits
+
+name: 'Delsarte-Goethals Codes'
+introduced: '\cite{doi:10.1016/0097-3165(75)90090-4}'
+
+description: 
+  - 'Delsarte-Goethals codes are a family of binary non-linear codes which were first introduced by Delsarte and Goethals in 1975. For even \(m >= 4\) and \( 0 \leq r \leq m/2-1\), the Delsarte-Goethals code with parameters \((m,r)\) is denoted by \(\mathcal{DG}(m,r)\) and has a lenght \(n= 2^m\), size \(N = 2^{r(m-1)+2m}\), and minimum distance \(d = 2^{m-1}-2^{m/2-1+r}\) \cite{doi:10.1016/0097-3165(75)90090-4}.'
```
Comment (AlexanderBarg) [inline codes/classical/bits/delsarte_goethals.yml:10]: "has a lenght" -> "has length"

File: codes/classical/bits/delsarte_goethals.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  - '\(\mathcal{DG}(m,r)\) is a subcode of the second order Reed-Muller code of length 2^m ( i.e. \(\mathcal{RM}(2,m)\)). Furthermore \(\mathcal{DG}(m,1)= \mathcal{RM}(2,m)\) . i.e for \(r=2\), \(\mathcal{DG}(m,1)\) codes are equivalent to \(\mathcal{RM}(2,m)\) codes of the same lenght.' # TTECC
+  - 'In fact, \(\mathcal{DG}(m,r)\) is the union of certain cosets of \(\mathcal{RM}(1,m)\) in \(\mathcal{RM}(2,m)\). Certain bilinear forms are used to specify these cosets \cite{doi:10.1016/0097-3165(75)90090-4}. This approach is used to explicitely construct the codes.  
+  - 'Kerdock code of lenght 2^m is a subcode of DG(m,r). Furthermore, DG(m,m/2) = K(m)' # TTECC
```
Comment (AlexanderBarg) [inline codes/classical/bits/delsarte_goethals.yml:13]: "length"

File: codes/classical/bits/delsarte_goethals.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  - '\(\mathcal{DG}(m,r+1)\) is a union of disjoint translates of \(\mathcal{DG}(m,r)\)' # TTECC
+  - 'While \(\mathcal{DG}(m,r)\) is not itself linear, it is the Gray map image of a certain linear code over \(\mathbb{Z_4}\) \cite{doi:10.1109/18.312154}. This is another way of constructing Delsarte-Goethals codes'
+  - 'In fact, the corresponding \(\mathbb{Z_4}\) codes are extended cyclic codes over \(\mathbb{Z_4}\). \cite{doi:10.1109/18.312154}'
+  - 'These codes are distance invariant \cite{doi:10.1109/18.312154}. Hence distance distribution and the weight distribution are the same.'
+  - 'Duals of \(\mathcal{DG}(m,r)\) exists in the sense that there exists codes with distance distribution equal to the McWillimas transform of the distance distribution of itself \cite{doi: 10.1016/0012-365X(90)90010-F}.'
```
Comment (AlexanderBarg) [inline codes/classical/bits/delsarte_goethals.yml:18]: "McWillimas" -> "MacWilliams"

File: codes/classical/bits/goethals.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+# cite The theory of error correcting codes - chapter 15.
+  
+  rate: '\(= {2^m -3m +1}/2^m\). Therefore, rate goes to 1 as the block length goes to infinity '
+  
+  decoders:
+    - 'Since the equivalent \(\mathbb{Z_4}\) codes are extended cyclic codes, efficient encoding-decoding is possible. \cite{doi:10.1109/18.312154} \cite{doi: 10.1109/18.476333}'. 
+
+
+relations:
+  parents:
```
Comment (AlexanderBarg) [inline codes/classical/bits/goethals.yml:29]: 2nd order Reed-Muller codes, Z4 linear codes

File: codes/classical/bits/goethals.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    - code_id: bits_into_bits
+      detail: ''
+  cousins:
```
Comment (AlexanderBarg) [inline codes/classical/bits/goethals.yml:32]: Preparata codes, Goethals-Delsarte codes

File: codes/classical/bits/goethals.yml [hunk 6]
```diff
@@ -0,0 +1,43 @@
+
+code_id: delsarte_goethals
```
Comment (valbert4) [inline codes/classical/bits/goethals.yml:2]: goethals

File: codes/classical/bits/goethals.yml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  
+  decoders:
+    - 'Since the equivalent \(\mathbb{Z_4}\) codes are extended cyclic codes, efficient encoding-decoding is possible. \cite{doi:10.1109/18.312154} \cite{doi: 10.1109/18.476333}'. 
+
+
+relations:
+  parents:
+    - code_id: bits_into_bits
+      detail: ''
+    - code_id: z_4_linear
```
Comment (valbert4) [inline codes/classical/bits/goethals.yml:32]: quaternary_over_z4

File: codes/classical/bits/nordstrom_robinson.yml [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+# Introduced-- An optimum nonlinear code -- doi:10.1016/S0019-9958(67)90835-2,,,,, N. V. Semakov, V. A. Zinov'ev, “Complete and Quasi-complete Balanced Codes”, Probl. Peredachi Inf., 5:2 (1969), 14–18; Problems Inform. Transmission, 5:2 (1969), 11–13
+
+# Construction -1 The theory of error correcting codes - chapter 11.
+# Construction -2 The Nordstrom-Robinson Code is the Binary Image of the Octacode
+# Orthogonal array specification -- http://mint.sbg.ac.at/desc_CNordstromRobinson.html#bib-CNordstromRobinson-3
+# 
+
+relations:
+  parents:
+    - code_id: kerdock
```
Comment (valbert4) [inline codes/classical/bits/nordstrom_robinson.yml:33]: Another student submitted kerdock and had this for nordstrom; please integrate this informaiton into a detail entry for kerdock: 'Specializes Kerdock code for \(m = 4\). Consequently, the Kerdock set is a set of 8 matrices in \(\mathbb(F)_{2}^{4X4}\). The resulting code is binary with parameters [16, 256, 6].'

File: codes/classical/bits/delsarte_goethals.yml [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    - 'Since the equivalent \(\mathbb{Z_4}\) codes are extended cyclic codes, efficient encoding-decoding is possible \cite{doi:10.1109/18.312154}.'
+
+realizations:
+  - 'Space-time signalling \cite{doi:10.1109/ICC.2004.1312536}.'
+
+relations:
+  parents:
+    - code_id: bits_into_bits
+      detail: ''
+    - code_id: z_4_linear
```
Comment (valbert4) [inline codes/classical/bits/delsarte_goethals.yml:35]: quaternary_over_z4

File: codes/classical/bits/delsarte_goethals.yml [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  
+  rate: '\(= {r(m-1)+2m}/2^m\). Therefore, rate goes to 0 as the block length goes to infinity. '
+  
+  decoders:
+    - 'Since the equivalent \(\mathbb{Z_4}\) codes are extended cyclic codes, efficient encoding-decoding is possible \cite{doi:10.1109/18.312154}.'
+
+realizations:
+  - 'Space-time signalling \cite{doi:10.1109/ICC.2004.1312536}.'
+
+relations:
```
Comment (valbert4) [inline codes/classical/bits/delsarte_goethals.yml:31]: For **all three** code entries, please provide detail statements for any cousin added, unless it is completely obvious why it's a cousin (this is very rare).

File: codes/classical/bits/nordstrom_robinson.yml (hunk 8 above)
Comment (valbert4) [reply codes/classical/bits/nordstrom_robinson.yml:33]: Add a detail for preparata as well

**Pull request closed without merging**